src/main/java/com/avispl/symphony/dal/device/core510i/QSYSCoreCommunicator.java

```diff
@@ -0,0 +1,311 @@
+		stats.put(QSYSCoreMonitoringMetric.DEVICE_ID.getName(), QSYSCoreConstant.NONE);
+		stats.put(QSYSCoreMonitoringMetric.DEVICE_NAME.getName(), QSYSCoreConstant.NONE);
+		stats.put(QSYSCoreMonitoringMetric.DEVICE_MODEL.getName(), QSYSCoreConstant.NONE);
+		stats.put(QSYSCoreMonitoringMetric.FIRMWARE_VERSION.getName(), QSYSCoreConstant.NONE);
+		stats.put(QSYSCoreMonitoringMetric.SERIAL_NUMBER.getName(), QSYSCoreConstant.NONE);
+	}
+
+	/**
+	 * Value of list statistics property of device info is NONE
+	 *
+	 * @param failedMonitor list statistics property
+	 */
+	private void updateDeviceInfoFailedMonitor(Map<String, String> failedMonitor) {
+		failedMonitor.put(QSYSCoreMonitoringMetricGroup.DEVICE_INFO.getName(), QSYSCoreConstant.GETTING_DEVICE_INFO_ERR);
+	}
+
+	/**
+	 * Retrieve device information
+	 *
+	 * @param stats list statistics property
+	 */
+	private void retrieveDeviceInfo(Map<String, String> stats) {
+		try {
+			if (this.loginInfo.getToken() != null) {
+				JsonNode response = doGet(buildDeviceFullPath(QSYSCoreURL.BASE_URI + QSYSCoreURL.DEVICE_INFO), JsonNode.class);
+
+				if (!response.get(QSYSCoreConstant.DATA).isEmpty()) {
+					ObjectMapper objectMapper = new ObjectMapper();
+					DeviceInfo deviceInfo = objectMapper.readValue(response.get(QSYSCoreConstant.DATA).toString(), DeviceInfo.class);
+
+					stats.put(QSYSCoreMonitoringMetric.DEVICE_ID.getName(), checkNoneData(deviceInfo.getDeviceId()));
+					stats.put(QSYSCoreMonitoringMetric.DEVICE_NAME.getName(), checkNoneData(deviceInfo.getDeviceName()));
+					stats.put(QSYSCoreMonitoringMetric.DEVICE_MODEL.getName(), checkNoneData(deviceInfo.getDeviceModel()));
+					stats.put(QSYSCoreMonitoringMetric.FIRMWARE_VERSION.getName(), checkNoneData(deviceInfo.getFirmwareVersion()));
+					stats.put(QSYSCoreMonitoringMetric.SERIAL_NUMBER.getName(), checkNoneData(deviceInfo.getSerialNumber()));
+				} else {
+					contributeNoneValueForDeviceStatistics(stats);
+					updateDeviceInfoFailedMonitor(failedMonitor);
+				}
+			} else {
+				contributeNoneValueForDeviceStatistics(stats);
+				updateDeviceInfoFailedMonitor(failedMonitor);
+			}
+		} catch (Exception e) {
+			contributeNoneValueForDeviceStatistics(stats);
+			updateDeviceInfoFailedMonitor(failedMonitor);
+		}
+	}
+
+	/**
+	 * Value of list statistics property of device info is NONE
+	 *
+	 * @param failedMonitor list statistics property
+	 */
+	private void updateDeviceIPFailedMonitor(Map<String, String> failedMonitor) {
+		failedMonitor.put(QSYSCoreMonitoringMetricGroup.IP_ADDRESS.getName(), QSYSCoreConstant.GETTING_DEVICE_IP_ERR);
+	}
+
+	/**
+	 * Retrieve device IP address
+	 *
+	 * @param stats list statistics property
+	 */
+	private void retrieveDeviceIPAddress(Map<String, String> stats) {
+		try {
+			if (this.loginInfo.getToken() != null) {
+				JsonNode response = doGet(buildDeviceFullPath(QSYSCoreURL.BASE_URI + QSYSCoreURL.DEVICE_IP_ADDRESS), JsonNode.class);
```

Comment:
Consider using DTOs in scenarios like this one. It is not always necessary, probably it's not necessary here either, but having DTOs would also help to extend this functionality in future, troubleshoot etc. 
Sometimes, when the payload is rather small (like {"success": true} for instance). there's no much sense in creating DTOs, but whenever we have a chance - we may have them. 

Alternaltively, you may look into com.avispl.symphony.dal.aggregator.parser.AggregatedDeviceProcessor functionality. It's got options for extracting property maps from json, based on yml. It's not a requirement, but still may be something you might want to take a look at :)

Reply:
I fixed it